FMB/SortInference.cpp

```diff
@@ -252,9 +253,30 @@ void SortInference::doInference()
    Array<Stack<unsigned>> varPositions(c->varCnt());
    ZIArray<unsigned> varsWithPosEq(c->varCnt());
    IntUnionFind localUF(c->varCnt()+1); // +1 to avoid it being 0.. last pos will not be used
+
+   // This set will be used to collect the variables that occur as part of an equality in every literal (implicitly becomes empty if there's a non-equality literal).
+   // We will use this information to detect maximum model sizes as any claues which is just {X = t_i} means that the sort of X is at most i
+   // We start by assuming that all variables are always in all equalities and then remove variables that are not
+   Set<unsigned> vars_always_in_equalities;
```

Comment:
Isn't it an overkill to use a set?

Yes, for the first literal, we don't know what the variable is. We could learn it there and check it's the same later on, couldn't we?

Reply:
Probably, yes. In the first literal we might have two possibilities but we get one or zero possibilities by the second literal so we only need two variables, not a set. I'll refactor after discussing the next comment.

Reply:
Great, sounds good!